### ISSUE TITLE
Release 5.8.4 not recommended - expect 5.8.5 soon

### ISSUE THREAD

**Issue by dbwiddis** [oshi/oshi#1794]
I went through the process of creating release 5.8.4 today.  Unfortunately once you release something to the Central Repository you can't delete it, so it's out there.   I'm already working on a 5.8.5, so don't hurry to update.

**TLDR:**  

'll be removing the `FsstatTest` introduced in #1743, and all dependencies on Mockito.  And an accidental dependency on Apache Commons StringUtils in #1747 that works when we have the codeassert dependency, but we remove that for Java 11 due to incompatibilities.

**Details:**

FYI: @squid-ney and @iamdebojyoti

There was a recent discussion in #1743 and #1746 about adding a dependency on Mockito.  

I went ahead and accepted it in #1743 without doing a careful enough review.
 - `mockito-core` was not properly scoped as a test dependency, so it's a compile-time dependency in the latest release, even though it's not used
 - there's an unknown incompatibility between the mockito dependency tree and our current Junit testing, when testing in Java 11.  I don't know what it is, and I'm not going to spend any time debugging it.  
 - Mockito's dependency tree includes a dependency on Objenesis, which uses classes in sun.misc.Unsafe which I'd prefer not to use without good reason.  A single test class using the dependency is not a good enough reason in my book. :)

#1746 still hasn't been merged, but is experiencing test failures that I don't understand.   

Bottom line: If we are going to include Mockito, it's probably going to be part of a complete rewrite of our testing system, keeping modularity in mind.  And that complete rewrite will be done by someone other than me.

**Comment by dbwiddis** [comment]
OK, so after removing Mockito I still had the issues, and realized it was many of the new tests incorrectly using `org.junit.Test` instead of the Jupiter API version.   Gah.  I hate modules.   So I might try putting that back.  Later, though.  Too many wasted hours tonight.

**Comment by mprins** [comment]
>  new tests incorrectly using org.junit.Test

I think we should add an enforcer rule to prevent `junit:junit` sneaking it's way in, Ive done some junt 4 to 5 migrations and found out that the transitive dependency on `junit:junit` in many projects is not properly scoped and even when it is an ide (and maven) will pick up and use it without qualm
Having an enforcer rule can prevent that by early failure - making it necessary to add proper excludes to the dependency

**Comment by hazendaz** [comment]
Use this -> https://github.com/skuzzle/restrict-imports-enforcer-rule, works well to block the import usage.  You could also block junit4 as well as noted.

**Comment by mprins** [comment]
Pretty sure I fixed all my problems using just the simple rule (without adding another dependency):
```xml
        <bannedDependencies>
            <excludes>
                <exclude>junit:junit</exclude>
                <exclude>org.hamcrest:hamcrest-core</exclude>
                <exclude>commons-lang:commons-lang</exclude>
            </excludes>
        </bannedDependencies>
```

**Comment by hazendaz** [comment]
I'd use both only because someone could shade junit (if not already in the wild) and someone could have used some different group id for unknown reasons that is just really junit 4.  The main idea on both is to prevent known GAV and the import block is to prevent even having that come in to look at.  I'm sure it would be hard to get a shaded one in but a transient could cause that pretty transparently.

**Comment by mprins** [comment]
I think code inspection is better done using checkstyle, eg. using https://checkstyle.sourceforge.io/config_imports.html#ImportControl
as that will catch class use and isn't limited to `imports`  as noted in https://github.com/skuzzle/restrict-imports-enforcer-rule#limitation

**Comment by dbwiddis** [comment]
> I think code inspection is better done using checkstyle

Speaking of which, see #1797 :)

**Comment by dbwiddis** [comment]
I'm still on the fence about brining back Mockito, even if it wasn't the cause of this issue.  I think my position remains that I'm happy to include it if it's used in more than one testcase, and appreciate the work done so far by those who have attempted to add it.  But really we probably need a complete ground-up redesign of all our testing, with a module-friendly test design.

**Comment by dbwiddis** [comment]
> Use this -> https://github.com/skuzzle/restrict-imports-enforcer-rule, works well to block the import usage. You could also block junit4 as well as noted.

Is there a quick configuration option there to disallow all imports from transitive dependencies?   Basically the equivalent of JMPS needing a `requires`.  A `StringUtils.EMPTY` snuck into a test because commons-lang3 is one of (many) transitive dependencies of one of our dependencies.  JDK 8 didn't care.  JDK11 modular build identified the problem instantly.
